spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -87,6 +91,12 @@
 
 	private final AtomicLong lastNoMessageAlert = new AtomicLong();
 
+	private final AtomicBoolean containerStopping = new AtomicBoolean();
```

Comment:
I would say we don't need this flag - the `active` is fully enough to track the stopping status.
And `isRunning()` is good for external concurrent lifecycle controls.